cocotb/share/makefiles/simulators/Makefile.cvc

```diff
@@ -47,48 +47,39 @@ endif
 
 ifeq (, $(CMD))
     $(error "Unable to locate command >$(CMD_BIN)<")
-else
-    CVC_BIN_DIR := $(shell dirname $(CMD))
-    export CVC_BIN_DIR
 endif
 
-#only interpreted mode works for the moment
-CVC_ITERP ?= 1
+CVC_INTERP := $(filter $(EXTRA_ARGS),+interp)
 
-ifeq ($(CVC_ITERP),1)
-    CVC_ARGS    += +interp
-endif
+ifneq (, $(CVC_INTERP)) # Interpreted mode
+`
+$(COCOTB_RESULTS_FILE): $(VERILOG_SOURCES) $(CUSTOM_COMPILE_DEPS) $(CUSTOM_SIM_DEPS)
+	-@rm -f $(COCOTB_RESULTS_FILE)
 
-# Compilation phase
-$(SIM_BUILD)/sim.vvp: $(SIM_BUILD) $(VERILOG_SOURCES) $(CUSTOM_COMPILE_DEPS)
-	MODULE=$(MODULE) \
-	TESTCASE=$(TESTCASE) TOPLEVEL=$(TOPLEVEL) TOPLEVEL_LANG=$(TOPLEVEL_LANG) COCOTB_SIM=1 \
-        $(CMD) $(CVC_ARGS) +acc+2 -o $(SIM_BUILD)/sim.vvp +define+COCOTB_SIM=1 +loadvpi=$(LIB_DIR)/libcocotbvpi_modelsim:vlog_startup_routines_bootstrap $(COMPILE_ARGS) $(EXTRA_ARGS) $(VERILOG_SOURCES)
+	MODULE=$(MODULE) TESTCASE=$(TESTCASE) TOPLEVEL=$(TOPLEVEL) TOPLEVEL_LANG=$(TOPLEVEL_LANG) COCOTB_SIM=1 \
+	$(CMD) +loadvpi=$(LIB_DIR)/libcocotbvpi_modelsim:vlog_startup_routines_bootstrap $(COMPILE_ARGS) $(SIM_ARGS) $(EXTRA_ARGS) $(VERILOG_SOURCES) $(PLUSARGS)
+
+	$(call check_for_results_file)
 
-# Execution phase
-ifeq ($(CVC_ITERP),1)
-    $(COCOTB_RESULTS_FILE): $(SIM_BUILD)/sim.vvp
 else
-    $(COCOTB_RESULTS_FILE): $(SIM_BUILD)/sim.vvp $(CUSTOM_SIM_DEPS)
-	MODULE=$(MODULE) \
-        TESTCASE=$(TESTCASE) TOPLEVEL=$(TOPLEVEL) TOPLEVEL_LANG=$(TOPLEVEL_LANG) COCOTB_SIM=1 \
-        $(SIM_BUILD)/sim.vvp $(SIM_ARGS) $(EXTRA_ARGS) $(PLUSARGS)
-endif
+
+# Compilation phase
+$(SIM_BUILD)/cvcsim: $(SIM_BUILD) $(VERILOG_SOURCES) $(CUSTOM_COMPILE_DEPS)
+	$(CMD) +verbose -o $(SIM_BUILD)/cvcsim $(CVC_ARGS) +loadvpi=$(LIB_DIR)/libcocotbvpi_modelsim:vlog_startup_routines_bootstrap $(COMPILE_ARGS) $(EXTRA_ARGS) $(VERILOG_SOURCES)
```

Comment:
Any idea why we use the modelsim one?

Reply:
CVC is not very popular and at some point needed same ifdefs as modelsim. Not sure if need to make another library just for CV .

Reply:
Do you think I should change this?

Reply:
Ah, found what I was looking for in some removed code:

https://github.com/cocotb/cocotb/blob/b30cf07cc81a650b71548ce3e24d0820b6f7d802/cocotb/share/makefiles/Makefile.inc#L58-L61

I think a single comment here explaining that this is intentional would do.